jina/serve/executors/__init__.py

```diff
@@ -122,9 +122,15 @@ def get_function_with_schema(fn: Callable) -> T:
         # otherwise, infer from the doc parameter (since generator endpoints expect only 1 document as input)
         is_generator = getattr(fn, '__is_generator__', False)
         if not is_generator:
+            assert (
+                'doc' not in fn.__annotations__
+            ), f'Cannot specify the `doc` parameter if the endpoint {fn.__name__} is not a generator'
             docs_annotation = fn.__annotations__.get('docs', None)
         else:
             docs_annotation = fn.__annotations__.get('doc', None)
+            assert (
```

Comment:
make sure a method with await is not confused with a generator

Reply:
added more tests